### DISCUSSION TITLE
86Box Linux AppImage compilation vs a cmake 'regular' build from source compilation

### DISCUSSION THREAD

**Discussion by foxcode64** [86Box/86Box#4370] [General]
Something I have noticed recently to do with the 'Releases' AppImage Linux-x86_64 executable and doing a compilation of 86Box myself.

I'm emulating a standard Model 30 PS/2 286 machine in 86Box with a 287 fpu.  Using the classic MS-DOS 'debug' program I write the following small assembly code that uses the x87 fpu instructions:

```
finit
fbld tbyte ptr [0200]
fbstp tbyte ptr [0300]
ret
```

The above program just loads via a ten byte packed BCD number at location 0200 onto the fpu stack ST(0) and then pops and stores it back at location 0300 in memory.  So at location 0200 we could have the BCD ten bytes:

```
0200:  56 34 12 00 00 00 00 00 00 00
```
To load the number 123456 ... and then after running the above program at location 0300 we would have:

```
0300:  56 34 12 00 00 00 00 00 00 00
```

If I run 86Box via a x86_Linux precompiled binary executable AppImage I find that my above program doesn't work.  I get at location 0300:

```
0300:  00 00 00 00 00 00 00 00 00 00
```

OTOH, if I compile my own 'regular' 86Box from source, it works fine.

I wonder if there is a problem with how the C compiler/optimiser is generating the code for the cpu/fpu emulation (which to be fair is pretty complicated code for a compiler to compile/optimise) in the AppImages(?)

**Comment by OBattler** [comment]
How old is the AppImage? It may be from an earlier build where the FBSTP instruction was not yet implemented.

**Reply by OBattler** [reply]
In fact, looking at it, FBSTP was implemented in one of the 4.2 builds, so the release 4.1.1 AppImage isn't going to have them. Get the latest AppImage from our Jenkins: https://ci.86box.net .

**Comment by foxcode64** [comment]
Ahh ok, I naturally assumed that all the instructions would be implemented for the system architecture you are emulating! :smiley:

I think in this case maybe the issue is the FBLD instruction as you have a commit recently dealing with that.  Hopefully then the next AppImage release of 86Box fixes this issue :smiley:
